dao/request.py

```diff
@@ -1,23 +1,55 @@
 import datetime
 
 from config import db
+from dao.mixin import OutputMixin
 
 
-class Request(db.Model):
+class Request(OutputMixin, db.Model):
+    RELATIONSHIPS_TO_DICT = True
+    REQUEST_REQUIRED_PARAMS = ['supplyName', 'status', 'description', 'uid', 'did']
+
     rid = db.Column(db.Integer, primary_key=True)
     supplyName = db.Column(db.String(50), nullable=False)
     time = db.Column(db.DateTime, default=datetime.datetime.utcnow())
     status = db.Column(db.Boolean, nullable=False)
     description = db.Column(db.String(100))
     uid = db.Column(db.Integer, db.ForeignKey('user.uid'), nullable=False)
+    did = db.Column(db.Integer, db.ForeignKey('donation.did'), nullable=False)
+
+    def __init__(self, **kwargs):
+        self.supplyName = kwargs.get('supplyName')
+        self.status = kwargs.get('status')
+        self.description = kwargs.get('description')
+        self.uid = kwargs.get('uid')
+        self.did = kwargs.get('did')
+
+    def __repr__(self):
+        return self.supplyName
 
-    def getAllRequests(self):
-        return self.query.all()
+    @property
+    def pk(self):
+        return self.rid
+
+    @staticmethod
+    def get_all_requests():
+        return Request.query.all()
 
     @staticmethod
-    def getRequestById(request_id):
-        return Request.query.filter_by(rid=request_id)
+    def get_request_by_id(request_id):
+        return Request.query.filter_by(rid=request_id).first()
 
-    def create(self, user):
+    def create(self, user, donation):
         user.requests.append(self)
+        donation.requests.append(self)
```

Comment:
The parameters of user and donation, and appending them to the relationships are not necessary, slqalchemy will do that automatically, i.e. they can be eliminated.

Edit: This comment is more for the DB team.

Reply:
Resolved in the request branch.